### PR TITLE
add cheap/meets min way of handling crd/cm deletes

### LINKED ISSUE #23: watch / sync for share CRD, driver configmap
minimally need to make sure recreated if deleted 

in case of CRDs, do we worry about sabotage?

library-go based (configobserver, even though the driver watches and restarts on change), or vanilla shared informer?

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes #23 

/assign @coreydaley

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -185,6 +167,73 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+func ensureConfigurationConfigMapExists(ctx context.Context, kubeClient kubeclient.Interface) error {
+	cmData, err := generated.Asset("config_configmap.yaml")
+	if err != nil {
+		return fmt.Errorf("error occurred reading file 'config_configmap.yaml': %s", err)
+	}
+	configMap := &corev1.ConfigMap{}
+	if err := yaml.Unmarshal(cmData, configMap); err != nil {
+		return fmt.Errorf("error occurred unmarshalling file 'config_configmap.yaml': %s", err)
+	}
+	_, err = kubeClient.CoreV1().ConfigMaps(defaultNamespace).Get(ctx, "csi-driver-shared-resource-config", metav1.GetOptions{})
```
Comment (coreydaley) [inline pkg/operator/starter.go:225]:
```suggestion
	_, err = kubeClient.CoreV1().ConfigMaps(defaultNamespace).Get(ctx, configMap.Name, metav1.GetOptions{})
```

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -185,6 +167,73 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+	if kerrors.IsNotFound(err) {
+		_, err = kubeClient.CoreV1().ConfigMaps(defaultNamespace).Create(ctx, configMap, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("error occurred creating 'csi-driver-shared-resource-config' ConfigMap: %s", err)
```
Comment (coreydaley) [inline pkg/operator/starter.go:229]:
```suggestion
			return fmt.Errorf("error occurred creating %q ConfigMap: %s", configMap.Name, err)
```

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -185,6 +167,73 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
...
+		}
+		klog.Infof("Successfully created ConfigMap %q", configMap.Name)
+	} else {
+		klog.Infof("ConfigMap '%q' already exists", configMap.Name)
+	}
+	if err != nil && !kerrors.IsNotFound(err) {
+		return fmt.Errorf("unexpected error determining if 'csi-driver-shared-resource-config' exists: %s", err)
```
Comment (coreydaley) [inline pkg/operator/starter.go:236]:
```suggestion
		return fmt.Errorf("unexpected error determining if %q exists: %s", configMap.Name, err)
```

**Review by coreydaley — CHANGES_REQUESTED** [review]

**Comment by gabemontero** [comment]
updated @coreydaley thanks

**Comment by gabemontero** [comment]
/assign @prietyc123 

for qe approved label

/assign @rolfedh 

for docs approved label

/assign @adambkaplan 

for px approved label

**Comment by rolfedh** [comment]
@gabemontero Do these changes have any doc impacts?

**Comment by gabemontero** [comment]
> @gabemontero Do these changes have any doc impacts?

no doc impacts for this one @rolfedh .... this change is just reliability fixes

**Review by rolfedh — APPROVED** [review]

**Comment by gabemontero** [comment]
thanks @rolfedh, but your using github pr approve is not sufficient for us.  You need post a comment with

/label docs-approved

**Comment by gabemontero** [comment]
how about that ... I didn't know I had that power in this repo @rolfedh :-)

in the future, if you could use that prow command, that would be great

**Comment by gabemontero** [comment]
Priti is out sick

/assign @jitendar-singh 

for qe approved

**Comment by gabemontero** [comment]
> /label qe-approved

@adambkaplan - looks like @jitendar-singh does not have the power to apply qe-approved for this repo

He was able to do it for csi-driver-shared-resource-operator PRs.

However, in comparing https://github.com/openshift/release/blob/master/core-services/prow/02_config/openshift/csi-driver-shared-resource-operator/_pluginconfig.yaml and https://github.com/openshift/release/blob/master/core-services/prow/02_config/openshift/csi-driver-shared-resource/_pluginconfig.yaml I don't see the difference that explains how to get him that power.

Can you follow up as part of what you did to enable no-FF for this repo in the first place?

thanks

**Comment by adambkaplan** [comment]
Added this manually (looks like I do have that power). Will look into qe-approvers in openshift/release.

**Comment by adambkaplan** [comment]
Looks like there is no special permission for qe-approved. This isn't the case for backport-approved and cherrypick-approved.

My guess is that the prow bot didn't receive the event for one reason or another. Our webhook with GitHub is known to flake on occasion.

**Comment by gabemontero** [comment]
> Looks like there is no special permission for qe-approved. This isn't the case for backport-approved and cherrypick-approved.
> 
> My guess is that the prow bot didn't receive the event for one reason or another. Our webhook with GitHub is known to flake on occasion.

cool thanks @adambkaplan

**Pull request merged**